src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -133,53 +122,127 @@ public String toString() {
+              NodeConfigCacheReaderUtil.readHeapMaxSizeInBytes(appContext.getNodeConfigCache(), esNode);
+      this.desiredCacheMaxSizeInBytes = null;
+      setDefaultStepSize(cacheType);
+    }
+
+    private void setDefaultStepSize(ResourceEnum cacheType) {
+      // TODO: Move configuration values to rca.conf
+      // TODO: Update the step size to also include percentage of heap size along with absolute value
+      switch (cacheType) {
+        case FIELD_DATA_CACHE:
+          // Field data cache having step size of 512MB
+          this.stepSizeInBytes = (long) CacheUtil.getMBSizeInBytes(512);
+          break;
+        case SHARD_REQUEST_CACHE:
+          // Shard request cache step size of 512KB
+          this.stepSizeInBytes = (long) CacheUtil.getKBSizeInBytes(512);
+          break;
+        default:
+          assert false : "unrecognized cache type: " + cacheType.name();
+      }
+    }
 
-    final Long cacheMaxSize = NodeConfigCacheReaderUtil.readCacheMaxSizeInBytes(nodeConfigCache, esNode, cacheType);
-    final Long heapMaxSize = NodeConfigCacheReaderUtil.readHeapMaxSizeInBytes(nodeConfigCache, esNode);
+    public Builder coolOffPeriod(long coolOffPeriodInMillis) {
+      this.coolOffPeriodInMillis = coolOffPeriodInMillis;
+      return this;
+    }
 
-    if (cacheMaxSize == null || cacheMaxSize == 0 || heapMaxSize == null || heapMaxSize == 0) {
-      return false;
-    } else {
-      currentCacheMaxSizeInBytes = cacheMaxSize;
-      heapMaxSizeInBytes = heapMaxSize;
-      cacheUpperBoundInBytes = (long) (heapMaxSizeInBytes * cacheSizeUpperBound);
-      return true;
+    public Builder increase(boolean isIncrease) {
+      this.isIncrease = isIncrease;
+      return this;
     }
-  }
 
-  private void setStepSize() {
-    // TODO: Update the step size to also include percentage of heap size along with absolute value
-    // Field data cache having step size of 512MB
-    stepSizeInBytes.put(ResourceEnum.FIELD_DATA_CACHE, 512 * 1_024L * 1_024L);
+    public Builder desiredCacheMaxSize(long desiredCacheMaxSizeInBytes) {
+      this.desiredCacheMaxSizeInBytes = desiredCacheMaxSizeInBytes;
+      return this;
+    }
 
-    // Shard request cache step size of 512KB
-    stepSizeInBytes.put(ResourceEnum.SHARD_REQUEST_CACHE, 512 * 1_024L);
-  }
+    public Builder stepSizeInBytes(long stepSizeInBytes) {
+      this.stepSizeInBytes = stepSizeInBytes;
+      return this;
+    }
 
-  private long getStepSize(final ResourceEnum cacheType) {
-    return stepSizeInBytes.get(cacheType);
-  }
+    public Builder upperBoundThreshold(double upperBoundThreshold) {
+      this.upperBoundThreshold = upperBoundThreshold;
+      return this;
+    }
 
-  private void setDesiredCacheMaxSize(final long desiredCacheMaxSize) {
-    this.desiredCacheMaxSizeInBytes = Math.min(desiredCacheMaxSize, cacheUpperBoundInBytes);
+    public ModifyCacheMaxSizeAction build() {
+      // fail to read max size from node config cache
```

Comment:
nit: Update to `In case of failure to read max size from node config cache`

Reply:
Modified